### PR TITLE
feat: upgrade rc-select version and add demo to bring in filterSort f…

### PR DESCRIPTION
**Pull request by headwindz**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/26414

[Alphabetical sorting of options in a Select Component](https://github.com/ant-design/ant-design/issues/26414)

### 💡 Background and solution

Add a `filterSort` prop which deals with options sorting on search, to `rc-select` and upgrade `rc-select` version in `ant-design`

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   The Select component adds the function of sorting options in the filter mode.  |
| 🇨🇳 Chinese |  Select 组件新增筛选模式下选项排序功能。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/select/demo/search-sort.md](https://github.com/n0ruSh/ant-design/blob/select-with-sort/components/select/demo/search-sort.md)
[View rendered components/select/index.en-US.md](https://github.com/n0ruSh/ant-design/blob/select-with-sort/components/select/index.en-US.md)
[View rendered components/select/index.zh-CN.md](https://github.com/n0ruSh/ant-design/blob/select-with-sort/components/select/index.zh-CN.md)

### CODE REVIEW

File: components/select/index.zh-CN.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/_0XzgOis7/Select.svg
 | dropdownRender | 自定义下拉框内容 | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | 下拉菜单的 style 属性 | CSSProperties | - |  |
 | filterOption | 是否根据输入项进行筛选。当其为一个函数时，会接收 `inputValue` `option` 两个参数，当 `option` 符合筛选条件时，应返回 true，反之则返回 false | boolean \| function(inputValue, option) | true |  |
+| filterSort | 搜索时对筛选结果项的排序函数, 类似[Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)里的compareFunction | (optionA:Option, optionB: Option) => number | - |  |
```
Comment (xrkffgg) [inline components/select/index.zh-CN.md:42]: 新的 api 吗？ 版本写上

File: components/select/demo/search-sort.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+order: 66
```
Comment (xrkffgg) [inline components/select/demo/search-sort.md:2]: 可以放到搜索的 Demo 附近

Comment (xrkffgg) [inline components/select/demo/search-sort.md:2]: 可以放到搜索的 Demo 附近

File: components/select/index.zh-CN.md (hunk 1 above)
Comment (hengkx) [reply components/select/index.zh-CN.md:42]: 4.9.0

File: components/select/demo/search-sort.md (hunk 2 above)
Comment (n0ruSh) [reply components/select/demo/search-sort.md:2]: 这里的order逻辑不太理解。例如搜索 Demo的order是1， 前面的 basic.md (order为0) 跟 后面的 multiple.md (order为2)，如果是放 搜索Demo附近的话，是不是要设置order 为2， 然后把现有的 order顺延, 例如 multiple.md 改成 order 3。 原来order为3的改成order为4这样？

Comment (xrkffgg) [reply components/select/demo/search-sort.md:2]: 写 4 ，其他不动

**Comment by afc163** [comment]
You should push force after rebase feature branch.

**Comment by n0ruSh** [comment]
> You should push force after rebase feature branch.

Yep. I accidentally rebased against upstream/master... :joy:

File: components/select/index.en-US.md [hunk 3]
```diff
@@ -38,6 +38,7 @@ Select component to select value from options.
 | dropdownRender | Customize dropdown content | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | The style of dropdown menu | CSSProperties | - |  |
 | filterOption | If true, filter options by input, if function, filter options against it. The function will receive two arguments, `inputValue` and `option`, if the function returns `true`, the option will be included in the filtered set; Otherwise, it will be excluded | boolean \| function(inputValue, option) | true |  |
+| filterSort | Sort function for search options sorting, see [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)'s compareFunction. | (optionA:Option, optionB: Option) => number | - | 4.9.0 |
```
Comment (xrkffgg) [inline components/select/index.en-US.md:41]:
```suggestion
| filterSort | Sort function for search options sorting, see [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)'s compareFunction | (optionA:Option, optionB: Option) => number | - | 4.9.0 |
```

File: components/select/demo/search-sort.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+---
+order: 4
+title:
+  zh-CN: 带排序的搜索
+  en-US: Search with sort
+---
+
+## zh-CN
+
+在搜索模式下对过滤结果项进行排序
```
Comment (xrkffgg) [inline components/select/demo/search-sort.md:10]:
```suggestion
在搜索模式下对过滤结果项进行排序。
```

英文的你 也加一下__

File: components/select/index.en-US.md [hunk 5]
```diff
@@ -38,6 +38,7 @@ Select component to select value from options.
 | dropdownRender | Customize dropdown content | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | The style of dropdown menu | CSSProperties | - |  |
 | filterOption | If true, filter options by input, if function, filter options against it. The function will receive two arguments, `inputValue` and `option`, if the function returns `true`, the option will be included in the filtered set; Otherwise, it will be excluded | boolean \| function(inputValue, option) | true |  |
+| filterSort | Sort function for search options sorting, see [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)'s compareFunction | (optionA:Option, optionB: Option) => number | - | 4.9.0 |
```
Comment (xrkffgg) [inline components/select/index.en-US.md:41]:
```suggestion
| filterSort | Sort function for search options sorting, see [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)'s compareFunction | (optionA: Option, optionB: Option) => number | - | 4.9.0 |
```

cn 同

**Review by xrkffgg — APPROVED** [review]

**Pull request merged**